### ISSUE TITLE
SignUp Wizard: Roles no longer link to next partition

### ISSUE THREAD

**Issue by raufrecht** [secc/RockPlugins#44]
I upgraded from 1.2 to 1.3 and now when I select my campus, it slides to the Roles, but the roles do not link to the next partition which is schedules.

**Comment by raufrecht** [comment]
I see in the code there is an if statement that now doesn't show the link if Role.Limit isn't greater than 0, what about if it is unlimited?

**Comment by smross** [comment]
Randy, what configuration are you using to experience this issue? I may be experiencing something similar, so would be curious to know.

What I'm doing (which may experience the same limitation):

- 1st partition of Campus
- 2nd partition of a Defined Type. In my case, there are 3 Defined Values for the Defined Type, and this Defined Type is also configured as a Group Member Attribute for the Group Type we are using. Parameter Name in SignupWizard matches the Key in the Defined Type
- 3rd partition of the Group Type Role.

If I don't put a total >=1 (i.e. I leave it blank for unlimited), then partition 2 is not a clickable link to partition 3 **when using the SignupWizard** mode. When using the _Genius_ or _CardPage_ modes it works as expected.

**Comment by raufrecht** [comment]
I have mine setup a little differently:

   - 1st partition of Campus
   - 2nd partition of Group Type Role
   - 3rd partition of Schedule

I updated the cardrole.lava to this:
                {%- assign role = partition -%}
{%- assign spotsLeft = role.Limit | Minus: role.Filled -%}
<div class="col-sm-6 col-md-4">
{% if spotsLeft > 0 and role.Limit > 0 %}
{% if partitionSize > 0 %}
<a href="#" onclick="doSlide('#{{divId}}_{{role.Value}}'); return false;">
{% else %}
<a href="{{partition.Url}}&GroupId={{partition.GroupId}}">
{% endif %}
{% elseif role.Limit == null %}
<a href="#" onclick="doSlide('#{{divId}}_{{role.Value}}'); return false;">
                    {% endif %}
<div class="serve-card" style="height: 180px;">
                        <div class="box text-center">
<div class="clearfix">
{% if role.Limit == null %}<span class="label label-default pull-right"
style="margin: 10px">{% if role.Limit == null %}{% else %}{% if spotsLeft
<= 0 %}Full{% else %}{{spotsLeft}} {{ 'Spot' |
PluralizeForQuantity:spotsLeft }} Remaining{% endif %}{% endif %}</span>{%
endif %}
</div>
{% assign icon = role.Entity | Attribute:"Icon" %}
{% if icon != '' and icon != empty %}
<i class="{{icon}} fa-5x"></i>
{% endif %}
                            <h3>{{ role.Entity.Name }}</h3>
                        </div>
</div>
{% if spotsLeft > 0 and role.Limit > 0 %}
</a>
{% endif %}
                </div>
Randy Aufrecht
IT Manager | 909-592-2282 x147

[image: Christ's Church of the Valley]
[image: Connect with us on the Web] <http://www.ccvsocal.com/>[image:
Facebook] <https://facebook.com/ccvsocal>[image: Instagram]
<http://www.instagram.com/ccvsocal>

**Comment by cbump** [comment]
Hey @raufrecht do you have a link to the actual signup that I can see and help troubleshoot?

**Comment by raufrecht** [comment]
I already fixed mine but it is at https://my.ccvsocal.com/page/802

Randy Aufrecht
IT Manager | 909-592-2282 x147

[image: Christ's Church of the Valley]
[image: Connect with us on the Web] <http://www.ccvsocal.com/>[image:
Facebook] <https://facebook.com/ccvsocal>[image: Instagram]
<http://www.instagram.com/ccvsocal>

**Comment by cbump** [comment]
Gotcha.  So the issue was related to the unlimited Roles condition?

**Comment by raufrecht** [comment]
Yes, if you use unlimited roles, it doesn't link. The original code didn't
have the condition for null.

Randy Aufrecht
IT Manager | 909-592-2282 x147

[image: Christ's Church of the Valley]
[image: Connect with us on the Web] <http://www.ccvsocal.com/>[image:
Facebook] <https://facebook.com/ccvsocal>[image: Instagram]
<http://www.instagram.com/ccvsocal>

**Comment by MichaelAllen** [comment]
I just ran into this same issue today. It was a simple fix, though. Just a missing null check as Randy said above.

I added a null check (`or role.Limit == null`) to the end of these 2 lines in CardWizard/CardRole.lava and that fixed the problem for us. 
https://github.com/secc/RockPlugins/blob/7f68e6f52800eb08aaae7293146bd69192f47c34/Plugins/org.secc.Connection/org_secc/Connection/CardWizard/CardRole.lava#L4
https://github.com/secc/RockPlugins/blob/7f68e6f52800eb08aaae7293146bd69192f47c34/Plugins/org.secc.Connection/org_secc/Connection/CardWizard/CardRole.lava#L23

I assume a similar change will be needed in the other CardWizard/Card____.lava files as well since they seem to be using that same conditional. I'll work on testing this change in those other lava files, then I'll send over a PR.
